### PR TITLE
feat(install): same install for dna and rna

### PR DESCRIPTION
**Pull request by jemten**
### This PR adds | fixes:

The installation process now installs the dependencies for all  pipelines into one conda environment per default. It is possible to select to only install the dependencies for one pipeline by specifying '--pipelines rd_dna' on the command line. 

resolves #1508 

### How to test:

- Automatic and continuous test suite
1. Checkout branch feature/common_install
2. Run `perl mip install --envn D_mip --reference_dir ../mip_references`

### Expected outcome:
- [x] Installation, unit and integration test suite pass
- [x] Generate install script with all dependencies 

### Review:
- [ ] Code review
- [x] Tests pass

This [version](https://semver.org/) is a:
- [x] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [ ] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by jemten** [comment]
Test result: 
Screenshot 2020-05-08 at 11 34 13

File: README.md [hunk 1]
```diff
@@ -121,10 +121,10 @@ Below are instructions for installing MIP for analysis of rare diseases. Install
 $ git clone https://github.com/Clinical-Genomics/MIP.git
 $ cd MIP
 ```
-##### 2.Install required perl modules from cpan
+##### 2.Install required perl modules from cpan to a specified conda environment
 
 ```Bash
-$ bash mip_install_perl.sh -e [mip_rd-dna] -p [$HOME/miniconda3] 
+$ bash mip_install_perl.sh -e [mip_rd-dna] -p [$HOME/miniconda3]
```
Comment (henrikstranneheim) [inline README.md:127]:
```suggestion
$ bash mip_install_perl.sh -e [mip] -p [$HOME/miniconda3]
```

File: README.md [hunk 2]
```diff
@@ -135,18 +135,15 @@ $ perl t/mip_install.test
 
 ##### 4.Create the install instructions for MIP
 ```Bash
-$ perl mip install rd_dna
+$ perl mip install --environment_name [mip_rd-dna] --reference_dir [$HOME/mip_references]
```
Comment (henrikstranneheim) [inline README.md:138]:
```suggestion
$ perl mip install --environment_name [mip] --reference_dir [$HOME/mip_references]
```

File: README.md (hunk 1 above)
Comment (henrikstranneheim) [reply README.md:127]: I think we should clean-up the read and remove any mentioning of 2 different installation procedures.
This sentance should also change:
`Below are instructions for installing MIP for analysis of rare diseases. Installation of the RNA pipeline follows a similar syntax.`

File: documentation/README.md [hunk 3]
```diff
@@ -134,17 +134,15 @@ $ perl t/mip_install.test
 
 ##### 4.Create the install instructions for MIP
 ```Bash
-$ perl mip install rd_dna
+$ perl mip install --environment_name [mip_rd-dna] --reference_dir [$HOME/mip_references]
```
Comment (henrikstranneheim) [inline documentation/README.md:137]:
```suggestion
$ perl mip install --environment_name [mip] --reference_dir [$HOME/mip_references]
```

File: documentation/README.md [hunk 4]
```diff
@@ -120,7 +120,7 @@ Below are instructions for installing MIP for analysis of rare diseases. Install
 $ git clone https://github.com/Clinical-Genomics/MIP.git
 $ cd MIP
 ```
-##### 2.Install required perl modules from cpan
+##### 2.Install required perl modules from cpan to a specified conda environment
 
 ```Bash
 $ bash mip_install_perl.sh -e [mip_rd-dna] -p [$HOME/miniconda3]
```
Comment (henrikstranneheim) [inline documentation/README.md:126]:
```suggestion
$ bash mip_install_perl.sh -e [mip] -p [$HOME/miniconda3]
```

File: lib/MIP/Cli/Mip/Install.pm [hunk 5]
```diff
@@ -2,34 +2,84 @@ package MIP::Cli::Mip::Install;
...
 ## CPANM
 use autodie qw{ :all };
 use MooseX::App::Command;
 use Moose::Util::TypeConstraints;
 use MooseX::Types::Moose qw{ ArrayRef Bool HashRef Int Str };
+use MooseX::Types::Structured qw{ Dict Optional };
 
 ## MIPs lib/
-use MIP::Cli::Utils qw{ run };
+#use MIP::Cli::Utils qw{ run };
```
Comment (henrikstranneheim) [inline lib/MIP/Cli/Mip/Install.pm:22]:
```suggestion
```
Remove?

File: lib/MIP/Cli/Mip/Install.pm [hunk 6]
```diff
@@ -2,34 +2,84 @@ package MIP::Cli::Mip::Install;
...
+q{Generates an installation script (mip.sh), which is used for installation of the Mutation Identification Pipeline (MIP) for rare diseases.}
+);
 
-command_usage(q{install <pipeline>});
+command_usage(q{mip <install> [options]});
 
 ## Define, check and get Cli supplied parameters
 _build_usage();
 
+sub run {
```
Comment (henrikstranneheim) [inline lib/MIP/Cli/Mip/Install.pm:43]: Nice to see that you could reuse this a lot. 👍

File: lib/MIP/Recipes/Pipeline/Install.pm [hunk 7]
```diff
@@ -39,13 +39,13 @@ BEGIN {
...
 
     # Functions and variables which can be optionally exported
-    our @EXPORT_OK = qw{ pipeline_install_rd_dna };
+    our @EXPORT_OK = qw{ pipeline_install };
 }
 
-sub pipeline_install_rd_dna {
+sub pipeline_install {
 
 ## Function : Install recipes for rd_dna pipeline
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Pipeline/Install.pm:50]:
```suggestion
## Function : Install recipes for mip pipelines
```

File: t/mip_install.test [hunk 8]
```diff
@@ -135,54 +135,42 @@ GetOptions(
...
+    }
 
-        my ( $success, $error_message, $full_buf, $stdout_buf, $stderr_buf ) = run(
-            command => \@mode_cmds,
-            verbose => $verbose
-        );
+    push @mode_cmds, $installation_mode;
 
-        ok( $success, qq{Executed $pipeline install $installation_mode} );
+    my ( $success, $error_message, $full_buf, $stdout_buf, $stderr_buf ) = run(
```
Comment (henrikstranneheim) [inline t/mip_install.test:165]: Would be great if you could use the child_process module here instead.

File: templates/mip_install_config.yaml [hunk 9]
```diff
@@ -42,6 +55,8 @@ singularity:
   chanjo:
     executable:
       chanjo:
+    pipeline:
+      - rd_dna
```
Comment (henrikstranneheim) [inline templates/mip_install_config.yaml:59]: Should these 2 lines be here? They seem out of place

File: README.md (hunk 1 above)
Comment (jemten) [reply README.md:127]: I agree the README is due for an overhaul

File: lib/MIP/Cli/Mip/Install.pm (hunk 6 above)
Comment (jemten) [reply lib/MIP/Cli/Mip/Install.pm:43]: Yeah it was easier than I first thought it would be.

File: templates/mip_install_config.yaml (hunk 9 above)
Comment (jemten) [reply templates/mip_install_config.yaml:59]: That was an earlier idea I had on how to merge the installation process. Forgot to remove them on some containers it seems.

**Comment by jemten** [comment]
> Fantastic work Anders!

Thanks!

File: t/mip_install.test (hunk 8 above)
Comment (jemten) [reply t/mip_install.test:165]: 👍

**Comment by jemten** [comment]
ready for review

File: documentation/README.md [hunk 10]
```diff
@@ -188,13 +189,13 @@ You can always supply `mip [process] [pipeline] --help` to list all available pa
 
 Example usage:
 ```Bash
-$ mip analyse rd_dna case_3 --sample_ids 3-1-1A --sample_ids 3-2-1U --sample_ids 3-2-2U --star_with_recipe samtools_merge --config 3_config.yaml
+$ mip analyse rd_dna case_3 --sample_ids 3-1-1A --sample_ids 3-2-1U --sample_ids 3-2-2U --start_with_recipe samtools_merge --config 3_config.yaml
```
Comment (henrikstranneheim) [inline documentation/README.md:192]: 👍

**Pull request merged**